pkg/kubectl/cmd/describe.go

```diff
@@ -38,6 +39,7 @@ given resource.`,
 			err := RunDescribe(f, out, cmd, args)
 			cmdutil.CheckErr(err)
 		},
+		ValidArgs: kubectl.DescribableResources(),
```

Comment:
The list of describable resources is more accurately the set of `RESTMappings` for which `f.Describer` returns a valid input.  The distinction doesn't matter for kube directly, but for embedders the ability to have that list include all the valid types will be helpful.

I will admit that there is no direct path from here to there, since `f.Object()` will probably fail on invalid config at that point.  Perhaps an extension to the `Factory` to return all potentially valid RESTMappers?